src/controllers/FileDownloadController.ts

```diff
@@ -0,0 +1,38 @@
+import { Response } from 'express';
+import { OK } from 'http-status-codes';
+import { inject } from 'inversify';
+import { controller, httpGet, requestParam, response } from 'inversify-express-utils';
+import { BaseAsyncHttpController } from './BaseAsyncHttpController';
+
+import { loggerInstance } from 'app/middleware/Logger';
+import { FileTransferService } from 'app/service/FileTransferService';
+import { DOWNLOAD_FILE_URI } from 'app/utils/Paths';
+
+const template = 'download-file';
+
+@controller(DOWNLOAD_FILE_URI)
+export class FileDownloadController extends BaseAsyncHttpController {
+
+    constructor(@inject(FileTransferService) private readonly fileTransferService: FileTransferService) {
+        super();
+    }
+
+    @httpGet('/prompt/:fileId')
+    public renderPrompt(@response() res: Response, @requestParam('fileId') fileId: string): void {
```

Comment:
is the response decorator needed?

Reply:
yes I'm calling `res.render`

Reply:
I though we could just use `this.httpContext` for consistency but it is not massive issue although I would not like to start using `@response()` in more places than just here. We settled on using `httpContext`.

Reply:
I just used this because I'm not using the BaseController

Reply:
`BaseAsyncHttpController` extends `BaseHttpController` which has `httpContext`